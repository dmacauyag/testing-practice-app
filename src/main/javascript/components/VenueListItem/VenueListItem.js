import '../../../resources/styles/components/VenueListItem/VenueListItem.css'
import React from 'react'

const VenueListItem = ({venue, onVenueSelect}) => {
  return (
    <li
      className='venues-list__item'
      onClick={() => onVenueSelect(venue.id)}
    >
      <span className="venues-list__item-name">{venue.name}</span>
    </li>
  )
}

export default VenueListItem
