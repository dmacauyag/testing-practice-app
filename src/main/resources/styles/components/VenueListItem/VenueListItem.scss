.venues-list__item {
  color: rgba(0, 0, 0, 0.87);
  cursor: pointer;
  display: block;
  line-height: 16px;
  margin: 0px;
  outline: none;
  padding: 12px;
  transition: all 900ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;

  &:hover {
    background-color: rgba(0, 0, 0, 0.05);
  }
}
