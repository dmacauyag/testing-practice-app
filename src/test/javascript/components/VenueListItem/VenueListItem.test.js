import { shallow } from 'enzyme'
import VenueListItem from '../../../../main/javascript/components/VenueListItem/VenueListItem'
import React from 'react'
import renderer from 'react-test-renderer'

describe('the VenueListItem component', () => {
  const mockVenues = {
    name: 'test name',
    id: 1234
  }
  const onVenueSelectMock =jest.fn()
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <VenueListItem
        venue={mockVenues}
        key={1234}
        onVenueSelect={onVenueSelectMock}
      />
    )
  })

  it('should correctly render the VenueListItem component', () => {
    expect(wrapper).toBeDefined()
  })

  describe('VenueListItem onclick', () => {
    it('should call the onVenueSelectMock function when an li is clicked', () => {
      wrapper.find('.venues-list__item').simulate('click')
      expect(onVenueSelectMock).toHaveBeenCalledTimes(1)
    })
  })
})
