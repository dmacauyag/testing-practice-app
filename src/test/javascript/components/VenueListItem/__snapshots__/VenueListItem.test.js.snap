// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the VenueListItem component should correctly render the VenueListItem component 1`] = `
<div>
  <span
    className="venues-list__item"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    open={null}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "background": "none",
        "backgroundColor": null,
        "border": 10,
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "cursor": "pointer",
        "display": "block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 16,
        "fontWeight": "inherit",
        "lineHeight": "16px",
        "margin": 0,
        "muiPrepared": true,
        "outline": "none",
        "padding": 0,
        "position": "relative",
        "textDecoration": "none",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "verticalAlign": null,
      }
    }
    tabIndex={0}
  >
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <div
        style={
          Object {
            "marginLeft": 0,
            "muiPrepared": true,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "position": "relative",
          }
        }
      >
        test name
      </div>
    </div>
  </span>
</div>
`;
